python/packages/autogen-core/docs/src/user-guide/core-user-guide/design-patterns/multiple-agents-concurrently.ipynb

```diff
@@ -0,0 +1,313 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 122,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@type_subscription(topic_type=\"urgent\")\n",
+    "class UrgentProcessor(RoutedAgent):\n",
+    "    @message_handler\n",
+    "    async def on_task(self, message: Task, ctx: MessageContext) -> None:\n",
+    "        print(f\"Urgent processor starting task {message.task_id}\")\n",
+    "        await asyncio.sleep(1)  # Simulate work\n",
+    "        print(f\"Urgent processor finished task {message.task_id}\")\n",
+    "\n",
+    "\n",
+    "@type_subscription(topic_type=\"normal\")\n",
+    "class NormalProcessor(RoutedAgent):\n",
+    "    @message_handler\n",
+    "    async def on_task(self, message: Task, ctx: MessageContext) -> None:\n",
+    "        print(f\"Normal processor starting task {message.task_id}\")\n",
+    "        await asyncio.sleep(3)  # Simulate work\n",
+    "        print(f\"Normal processor finished task {message.task_id}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After registering the agents, we can publish messages to the \"urgent\" and \"normal\" topics:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 117,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Normal processor starting task normal-1\n",
+      "Urgent processor starting task urgent-1\n",
+      "Urgent processor finished task urgent-1\n",
+      "Normal processor finished task normal-1\n"
+     ]
+    }
+   ],
+   "source": [
+    "runtime = SingleThreadedAgentRuntime()\n",
+    "\n",
+    "await UrgentProcessor.register(runtime, \"urgent_processor\", lambda: UrgentProcessor(\"Urgent Processor\"))\n",
+    "await NormalProcessor.register(runtime, \"normal_processor\", lambda: NormalProcessor(\"Normal Processor\"))\n",
+    "\n",
+    "runtime.start()\n",
+    "\n",
+    "await asyncio.gather(\n",
+    "    runtime.publish_message(Task(task_id=\"normal-1\"), topic_id=TopicId(type=\"normal\", source=\"default\")),\n",
+    "    runtime.publish_message(Task(task_id=\"urgent-1\"), topic_id=TopicId(type=\"urgent\", source=\"default\")),\n",
+    ")\n",
+    "\n",
+    "await runtime.stop_when_idle()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Direct Messages\n",
+    "\n",
+    "Finally, in contrast to the previous patterns, this pattern is about direct messages. It demonstrates two ways to send messages:\n",
+    "\n",
+    "- Direct messaging between agents\n",
+    "- Sending messages from the runtime to specific agents\n",
+    "\n",
+    "Things to consider when using direct messaging:\n",
+    "- Messages are addressed using {py:class}`~autogen_core.components.AgentId`\n",
+    "- The sender can expect and receive a response from the target agent"
```

Comment:
Add to this to explain the worker agent instances are created by the runtime when a message is delivered to the specific agent id. Add reference to the core concept doc on agent life cycle.